lib/pbench/server/api/__init__.py

```diff
@@ -286,7 +286,27 @@ def post(self):
             app.logger.debug("Bad file extension received")
             abort(400, message="File extension not supported. Only .xz")
 
+        try:
+            content_length = int(request.headers.get("Content-Length"))
+        except ValueError:
+            abort(400, message="Invalid content-length header, not an integer")
+        except Exception:
+            abort(400, message="Missing required content-length header")
+        else:
+            if content_length > _MAX_UPLOAD_DATA_SIZE:
+                abort(
+                    400,
+                    message="Content object too large, keep it at 1 GB"
+                    f" ({content_length:d}) and  under",
+                )
+            elif content_length == 0:
+                abort(
+                    400,
+                    message="Content object is empty, please re-check the file contents",
```

Comment:
Consider the wording, "Payload was empty, nothing was uploaded".

Reply:
Also, worth keeping in mind here that we're checking the Content-Size header, not the actual payload: so while the caller claimed a length of 0, that doesn't actually mean the "object is empty", only that they said it was.